.github/workflows/devshell.yml

```diff
@@ -0,0 +1,174 @@
+name: CI @ devshell
+
+on: [pull_request, push]
+
+jobs:
+  general:
+    runs-on: ubuntu-18.04
+    container:
+      image: balabit/syslog-ng-devshell:latest
+      env:
+        PYTHONUSERBASE: /github/home/python_packages
+        PATH: /github/home/python_packages/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+        CONFIGURE_FLAGS:
+          --prefix=/github/home/install/syslog-ng
+          --enable-all-modules
+          --with-python=${{ matrix.python }}
+        CMAKE_FLAGS:
+          -DCMAKE_C_FLAGS=-Werror
+          -DCMAKE_INSTALL_PREFIX=/github/home/install/syslog-ng
+          -DPYTHON_VERSION=${{ matrix.python }}
+
+    strategy:
+      matrix:
+        python: [2, 3]
+        build-tool: [autotools, cmake]
+      fail-fast: false
+
+    steps:
+      - name: Checkout syslog-ng source
+        uses: actions/checkout@v1
```

Comment:
Please use `actions/checkout@v2` so that the reexecute button could work.

Reply:
Thanks!

Reply:
Wow it break the Light tests...
```
Built target pytest-self-check
Scanning dependencies of target pytest-linters
An error has occurred: FatalError: git failed. Is it installed, and are you in a Git repository directory?
Check the log at /github/home/.cache/pre-commit/pre-commit.log
```

Reply:
https://github.com/actions/checkout/issues/126

Reply:
This error happens, when the `git` running in the docker is older, than `2.18`.

`bionic` has `git` version `2.17`, focal has `2.25`. Maybe we can merge it with `v1`, and presumably we can change the `devshell` to `focal` soon, and update to `v2 `then.

Another alternative would be to install `2.18` in the `devshell`, but, I think it would be more painful.

What do you think?